### DISCUSSION TITLE
Possible to find TPP objects that don't have an association

### DISCUSSION THREAD

**Discussion by bmartin271** [Venafi/venafips#108] [Q&A] [answered]
Is it possible to find application objects that don't have an associated certificate or device objects that don't have an application.  I would like to identify and clean up orphans

**Comment by gdbarron** [comment, marked as answer]
For applications I would use `Find-TppObject -Attribute 'Certificate' -Pattern ''`.  This will return all items which have a certificate attribute with no value.  This might return objects other than applications so you'll just need to filter those out.  You'll need v4.2.4 for this.

For devices I'd get all devices with `Find-TppObject -Class 'Device'`, loop through each and run `Find-TppObject -Path <devicepath>`.  If the path search does not return any items, there are no associated applications.
